### ISSUE TITLE
`enum` struct types are not included in the metadata information.

### ISSUE THREAD

**Issue by vladilen11** [MystenLabs/sui#20205]
This is my contract:

`assets_account_status.move`
```move
  #[allow(unused_use)]
  
  /* Autogenerated file. Do not edit manually. */
  
  module merak::assets_account_status {

  public enum AccountStatus has copy, drop , store {
                                Liquid,Frozen,Blocked
                        }

  public fun new_liquid(): AccountStatus {
    AccountStatus::Liquid
  }

  public fun new_frozen(): AccountStatus {
    AccountStatus::Frozen
  }

  public fun new_blocked(): AccountStatus {
    AccountStatus::Blocked
  }
}
```

`assets_account.move`
```move
  #[allow(unused_use)]
  
  /* Autogenerated file. Do not edit manually. */
  
  module merak::assets_account {

  use std::ascii::String;

  use merak::assets_account_status::AccountStatus;

  use merak::assets_status::Status;

  public struct Account has copy, drop, store {
    balance: u64,
    status: AccountStatus,
  }

  public fun new(balance: u64, status: AccountStatus): Account {
    Account {
                                   balance,status
                               }
  }
}
```

I defined an enum type `AccountStatus` in the `assets_account_status` contract and a struct type in the `assets_account` contract which declares a `status` with type `AccountStatus`.
After deploying the contract, I found a detail that there is no `enum` related structure construction information in the metadata information of the contract.
This is the metadata information of this contract: https://gist.github.com/vladilen11/cba593fbbfdc6b060e0feeec77f2a27a

As you can see, I can query the construct information of Account: https://gist.github.com/vladilen11/cba593fbbfdc6b060e0feeec77f2a27a#file-merak_contract_metadata-json-L16-L44

But I can't query the struct information about the enum type `AccountStatus`.

**My Requirement:**
I am parsing out all the relevant bcs information of this contract through the contract's metadata/abi, this way I can directly parse out the bcs structure through the metadata's data on typescript, so that I don't need to assemble the bcs structure manually every time.

**Comment by tzakian** [comment]
Thanks for the report @vladilen11. Working on a fix for this right now which should hopefully be in the next release.

**Comment by vladilen11** [comment]
I don't think this issue, for now, can be closed, we need to finish syncing all the support on `ts-sdk`, `graphql` and other packages releated to close this issue, what do you think? @tzakian 

I create a pr for ts-sdk: https://github.com/MystenLabs/sui/pull/20480
